### DISCUSSION TITLE
How to get danish jetstream translations

### DISCUSSION THREAD

**Discussion by simonj** [Laravel-Lang/common#4] [Q&A]
Hi - Really awesome project.

Am a little bit confused about the publisher, translations-template, lang what all the difference is 🤔

But i manage to install this https://publisher.laravel-lang.com/installation/ but am missing the jetstream, and am not sure how to add the danish version of that.

I think it's the da.json file that is located in /lang/da.json but how come it's not a php file like the others?

**Comment by andrey-helldar** [comment]
Hi Simon!

The publisher itself does not contain translation files. It is just a mechanism for setting them up.

For your purposes, I recommend that you perform the following steps, given that you have already installed it:

```bash
composer remove --dev laravel-lang/publisher
composer require --dev laravel-lang/common
php artisan lang:add da
```

The `Laravel Lang: Common` makes it easy to install packages.

https://github.com/Laravel-Lang/common

**Reply by simonj** [reply]
Ahh okay, Awesome @andrey-helldar 

Do you know what i should do to get the Jetstream translation also?

**Reply by andrey-helldar** [reply]
I know) Just follow the commands I suggested and open the Jetstream page)

**Reply by simonj** [reply]
Oh, am using InertiaJS :( forgot to mention that

**Reply by andrey-helldar** [reply]
@SimonJ, the `laravel-lang/lang` package contains translations for Jetstream 1-3, but it cannot install them itself, so it needs `laravel-lang/publisher`.

By installing the `laravel-lang/common` package, it will automatically add all the necessary dependencies to your application.

After that you can [add](https://publisher.laravel-lang.com/using/add.html), [update](https://publisher.laravel-lang.com/using/update.html), [reset](https://publisher.laravel-lang.com/using/reset.html) and [remove](https://publisher.laravel-lang.com/using/remove.html) localizations using the publisher's console commands.

I also recommend adding the `scripts.post-update-cmd` section in the `composer.json` file:

```json
{
    "scripts": {
        "post-update-cmd": [
            "php artisan lang:update"
        ]
    }
}
```

This will automatically update the localization files with each installation, update or removal of dependencies.

**Reply by andrey-helldar** [reply]
> Oh, am using InertiaJS :( forgot to mention that

As far as I know, Jetstream takes translation files from the framework itself, so updating the files in it will also "update" Inertia.

**Reply by andrey-helldar** [reply]
Indeed, the translation does not work.

Now checked on a clean project. Livewire works because uses translations of the framework, and Inertia contains keys embedded in compiled files.

Livewire: https://github.com/laravel/jetstream/blob/294d97f12faaf1b3dac34508145eeaefbfc447c0/stubs/livewire/resources/views/auth/register.blade.php#L50-L52

Inertia: https://github.com/laravel/jetstream/blob/3.x/stubs/inertia/resources/js/Pages/Auth/Register.vue#L102-L104

I assume that for Inertia the ability to translate the interface was not developed by the Laravel team.

**Reply by simonj** [reply]
Thanks a lot @andrey-helldar I got the translation file. Then i will just have to use something like https://github.com/xiCO2k/laravel-vue-i18n and then update all Jetstream files with translate tag, i think :)

**Reply by andrey-helldar** [reply]
@SimonJ, I haven't worked much with Jetstream, so I can't fully help you with it, but if you can use this package, then that's good.

Our team is collaborating with its developer :)

**Reply by andrey-helldar** [reply]
@SimonJ, I found an accepted PR in Jetstream 2 for Inertia. You probably have the opportunity to use the same mechanism for the 3rd version.

See: https://github.com/laravel/jetstream/pull/467

**Reply by andrey-helldar** [reply]
Or no... https://github.com/laravel/jetstream/issues/731
